packages/react-server-gulp-module-tagger/README.md

```diff
@@ -0,0 +1,45 @@
+# react-server-gulp-module-tagger
+
+A [gulp](http://gulpjs.com) plugin for tagging [react-server](https://www.npmjs.com/package/react-server) logger instances with information about the module they're being used in.
+
+To transpile your React for use with [React Server](https://www.npmjs.com/package/react-server), install gulp and the plugin
+
+    npm i -D gulp react-server-gulp-module-tagger
+
+Then add the task to your gulpfile
+
+    const gulp = require('gulp');
+    const tagger = require('react-server-gulp-module-tagger');
+    gulp.task('compile', () => {
+        gulp.src('src')
+            .pipe(tagger())
+            .pipe(gulp.dest('dist'));
+    });
+
+Compile task might also use [Babel](https://babeljs.io) with the [React Server Babel preset](https://www.npmjs.com/package/babel-preset-react-server)
+
+    const gulp = require('gulp');
+    const babel = require('gulp-babel');
+    const tagger = require('react-server-gulp-module-tagger');
+
+    gulp.task('compile', () => {
+        gulp.src('src')
+            .pipe(tagger())
+            .pipe(babel({ presets: ['react-server'] }))
+            .pipe(gulp.dest('dist'));
+    });
+
+Given a [React Server](https://www.npmjs.com/package/react-server) getLogger call, adds the correct arguments to keep the server and the browser in sync.
+
+For example, given the following three files
+
+#### src/foo.js
+    const logger = require("react-server").logging.getLogger(__CACHE__);
+
+#### src/bar.js
+    let logger = require("react-server").logging.getLogger(__LOGGER__);
+
+#### src/baz.js
+    var logger = require("react-server").logging.getLogger(__CHANNEL__);
```

Comment:
As far as I can tell, we support `__CACHE__` and `__CHANNEL__` for historical, internal uses, like our ipc for our cluster.  I'm not sure if we plan to release them, but I gather we want to support arbitrary tokens, per #12.  

I would prefer to document `__CACHE` and `__CHANNEL__` though, in the meantime, since it would be pretty mystifying if someone were to stumble across them; I think my question is how.  My first intuition was to give an example using `__CACHE__` and `__CHANNEL__` with a destructuring assignment

```
let {name, color} = __CACHE__;
let {name, color: {server, client}} = __CHANNEL__;
```

but I'm not sure if we want to make the structure of the tags part of our public api.  The other option (if we decide to keep the api small by only documenting that "a module context is substituted") would be to document `__CACHE__` and `__CHANNEL__` as "reserved for future functionality", with the assumption that caching and clustering are forthcoming.  Which do you think is a better approach?